cmd/booster-http/server.go

```diff
@@ -194,6 +210,51 @@ func (s *HttpServer) handlePieceRequest(w http.ResponseWriter, r *http.Request)
 	}
 }
 
+func (s *HttpServer) handleBlockRequest(w http.ResponseWriter, r *http.Request) {
+	startTime := time.Now()
+	ctx, span := tracing.Tracer.Start(r.Context(), "http.block_cid")
+	defer span.End()
+
+	stats.Record(ctx, metrics.HttpBlockByCidRequestCount.M(1))
+
+	// Remove the path up to the payload cid
+	prefixLen := len(s.blockBasePath())
+	if len(r.URL.Path) <= prefixLen {
+		msg := fmt.Sprintf("path '%s' is missing block CID", r.URL.Path)
+		writeError(w, r, http.StatusBadRequest, msg)
+		stats.Record(ctx, metrics.HttpPayloadByCid400ResponseCount.M(1))
+		return
+	}
+
+	blockCidStr := r.URL.Path[prefixLen:]
+	blockCid, err := cid.Parse(blockCidStr)
+	if err != nil {
+		msg := fmt.Sprintf("parsing payload CID '%s': %s", blockCidStr, err.Error())
+		writeError(w, r, http.StatusBadRequest, msg)
+		stats.Record(ctx, metrics.HttpPayloadByCid400ResponseCount.M(1))
+		return
+	}
+
+	data, err := s.api.GetBlockByCid(ctx, blockCid)
+	if err != nil {
+		msg := fmt.Sprintf("server error getting data for block '%s': %s", blockCidStr, err.Error())
+		writeError(w, r, http.StatusInternalServerError, msg)
+		stats.Record(ctx, metrics.HttpBlockByCid500ResponseCount.M(1))
+		return
+	}
+
+	utf8Name := url.PathEscape(blockCidStr)
+	asciiName := url.PathEscape(regexp.MustCompile("[[:^ascii:]]").ReplaceAllLiteralString(blockCidStr, "_"))
```

Comment:
Instead of doing all the escaping, can you just use blockCid.String()?

Reply:
I might be wrong but I think this due to how Go treats a string as slice of bytes(runes). It would still try to use the special characters so forcing them to escape would be a safer choice.
BTW, I picked this directly from Kubo.
https://github.com/ipfs/kubo/blob/1d5e46ac97dbd9862818b191517a4dc04186a26e/core/corehttp/gateway_handler.go#L688-L692